### PR TITLE
Protect the kafka-profision setting behind a mutex

### LINKED ISSUE #1983: fatal error: concurrent map read and map write
**Describe the bug**
The error happens sometimes, i guess some process triggers that.
Please ignore `This field was missing!!!!!!!!` as it is our internal message.

Please find attached file with surrouding logs 

[On_demand_report_2022-07-05T10_54_28.180Z_d7538140-fc50-11ec-84ba-cf7f04fb0981.csv](https://github.com/jaegertracing/jaeger-operator/files/9045750/On_demand_report_2022-07-05T10_54_28.180Z_d7538140-fc50-11ec-84ba-cf7f04fb0981.csv)

### PR DESCRIPTION
**Pull request by iblancasa**
## Which problem is this PR solving?
- Continuation of #2278 to fix #1983

## Short description of the changes
- Protect the Kafka Operator integration with a mutex.

### CODE REVIEW

File: pkg/autodetect/operatorconfig.go [hunk 1]
```diff
@@ -112,3 +127,36 @@ func (c *operatorConfigurationWrapper) GetESPIntegration() ESOperatorIntegration
...
+func (c *operatorConfigurationWrapper) SetKafkaIntegration(e interface{}) {
+	var integration string
+	switch v := e.(type) {
+	case string:
+		integration = v
+	case KafkaOperatorIntegration:
+		integration = v.String()
+	default:
+		integration = KafkaOperatorIntegrationNo.String()
+	}
```
Comment (frzifus) [inline pkg/autodetect/operatorconfig.go:140]:
```suggestion
	integration := KafkaOperatorIntegrationNo.String()
	if v, ok := e.(string); ok {
		integration = v
	} else if v, ok := e.(KafkaOperatorIntegration); ok {
		integration = v.String()
	}
```

File: pkg/autodetect/operatorconfig.go [hunk 2]
```diff
@@ -39,6 +39,21 @@ func (p ESOperatorIntegration) String() string {
 	return [...]string{"Yes", "No"}[p]
 }
 
+// KafkaOperatorIntegration holds the if the Kafka Operator integration is enabled.
+type KafkaOperatorIntegration int
+
+const (
+	// KafkaOperatorIntegrationYes represents the Kafka Operator integration is enabled.
+	KafkaOperatorIntegrationYes KafkaOperatorIntegration = iota
```
Comment (frzifus) [inline pkg/autodetect/operatorconfig.go:47]: Does it make sense to start with `KafkaOperatorIntegrationUnknown`? Otherwise default is yes?

Comment (iblancasa) [reply pkg/autodetect/operatorconfig.go:47]: Default is no

File: pkg/autodetect/operatorconfig.go (hunk 1 above)
Comment (iblancasa) [reply pkg/autodetect/operatorconfig.go:140]: I would prefer to follow the current approach. Is less "hacky" and more explicit.

**Review by frzifus — CHANGES_REQUESTED** [review]
Looks like you messed up your branch/history

**Comment by iblancasa** [comment]
> Looks like you messed up your branch/history

Fixed. I didn't sign off one empty commit.

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**